app/models/user.rb

```diff
@@ -15,20 +15,34 @@ class User < ApplicationRecord
 
   attr_accessor :remember_token, :reset_token
 
-  has_many :awards
-  has_many :submissions
-  has_many :participations
+  has_many :submissions, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :notifications, dependent: :destroy
+  # user.followers will return all users that follow THIS user
+  has_many :followers, class_name: 'Follower', foreign_key: 'following_id', dependent: :destroy
+  # user.following will return all users that THIS user follows
+  has_many :following, class_name: 'Follower', dependent: :destroy
+  has_many :user_session, dependent: :nullify
+
+  has_many :challenge_entries, dependent: :destroy
+  has_many :participations, dependent: :destroy
+  has_many :awards, dependent: :destroy
   has_many :badges, through: :awards
   has_many :challenges, through: :participations
-  has_many :comments
-  has_many :moderator_logs
-  has_many :moderator_logs, as: :target
-  has_one :moderator_application
+  has_many :created_challenges,
+           class_name: 'Challenge', foreign_key: 'creator_id', dependent: :nullify
+  has_many :user_feedbacks, dependent: :nullify
```

Comment:
Just for my sanity, does this work when there's a foreign_key relation on user_id in user_feedback?

Reply:
ruby handles foreign keys when calling destroy and I can confirm it does work when i run the dev_tasks scripts.

Reply:
Yes. Rails handles the foreign key stuff. Run the dev script if you want to see it.